mongodbatlas/fw_resource_mongodbatlas_search_node.go

```diff
@@ -0,0 +1,361 @@
+		return
+	}
+	if err := waitSearchNodeDelete(ctx, projectID, clusterName, connV2, deleteTimeout); err != nil {
+		resp.Diagnostics.AddError("error during search node delete", err.Error())
+		return
+	}
+}
+
+func (r *SearchNodeRS) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	projectID, clusterName, err := splitSearchNodeImportID(req.ID)
+	if err != nil {
+		resp.Diagnostics.AddError("error splitting search node import ID", err.Error())
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("project_id"), projectID)...)
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("cluster_name"), clusterName)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+func splitSearchNodeImportID(id string) (projectID, clusterName string, err error) {
+	var re = regexp.MustCompile(`(?s)^([0-9a-fA-F]{24})-(.*)$`)
+	parts := re.FindStringSubmatch(id)
+
+	if len(parts) != 3 {
+		err = errors.New("use the format {project_id}-{cluster_name}")
+		return
+	}
+
+	projectID = parts[1]
+	clusterName = parts[2]
+	return
+}
+
+func waitSearchNodeStateTransition(ctx context.Context, projectID, clusterName string, client *admin.APIClient, timeout time.Duration) (*admin.ApiSearchDeploymentResponse, error) {
+	stateConf := &retry.StateChangeConf{
+		Pending: []string{retrystrategy.RetryStrategyPausedState, retrystrategy.RetryStrategyPlanningState, retrystrategy.RetryStrategyWorkingState,
+			retrystrategy.RetryStrategyDeletedState, retrystrategy.RetryStrategyPendingState},
+		Target:     []string{retrystrategy.RetryStrategyIdleState},
+		Refresh:    searchNodeRefreshFunc(ctx, projectID, clusterName, client),
+		Timeout:    timeout,
+		MinTimeout: 1 * time.Minute,
+		Delay:      1 * time.Minute,
+	}
+
+	result, err := stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+	if deploymentResp, ok := result.(*admin.ApiSearchDeploymentResponse); ok && deploymentResp != nil {
+		return deploymentResp, nil
+	}
+	return nil, errors.New("did not obtain valid result when waiting for search node state transition")
+}
+
+func waitSearchNodeDelete(ctx context.Context, projectID, clusterName string, client *admin.APIClient, timeout time.Duration) error {
+	stateConf := &retry.StateChangeConf{
+		Pending: []string{retrystrategy.RetryStrategyIdleState, retrystrategy.RetryStrategyPausedState, retrystrategy.RetryStrategyPlanningState,
+			retrystrategy.RetryStrategyWorkingState, retrystrategy.RetryStrategyPendingState},
+		Target:     []string{retrystrategy.RetryStrategyDeletedState},
+		Refresh:    searchNodeRefreshFunc(ctx, projectID, clusterName, client),
+		Timeout:    timeout,
+		MinTimeout: 30 * time.Second,
+		Delay:      1 * time.Minute,
+	}
+	_, err := stateConf.WaitForStateContext(ctx)
+	return err
+}
+
+func searchNodeRefreshFunc(ctx context.Context, projectID, clusterName string, client *admin.APIClient) retry.StateRefreshFunc {
```

Comment:
you will likely run into my same problem that I faced [here](https://github.com/mongodb/mongodbatlas-cloudformation-resources/pull/810/files) but you should unit test this by mocking the client